src/Microsoft.Diagnostics.Monitoring.WebApi/Metrics/MetricsStore.cs

```diff
@@ -45,18 +46,49 @@ public override bool Equals(object obj)
 
         private Dictionary<MetricKey, Queue<ICounterPayload>> _allMetrics = new Dictionary<MetricKey, Queue<ICounterPayload>>();
         private readonly int _maxMetricCount;
+        private ILogger<MetricsStoreService> _logger;
 
-        public MetricsStore(int maxMetricCount)
+        private HashSet<string> _observedErrorMessages = new();
+        private HashSet<(string provider, string counter)> _observedEndedCounters = new();
+
+        public MetricsStore(ILogger<MetricsStoreService> logger, int maxMetricCount)
         {
             if (maxMetricCount < 1)
             {
                 throw new ArgumentException(Strings.ErrorMessage_InvalidMetricCount);
             }
             _maxMetricCount = maxMetricCount;
+            _logger = logger;
         }
 
         public void AddMetric(ICounterPayload metric)
         {
+            if (metric is PercentilePayload payload && !payload.Quantiles.Any())
+            {
+                // If histogram data is not generated in the monitored app, we can get Histogram events that do not contain quantiles.
+                // For now, we will ignore these events.
+                return;
+            }
+            //Do not accept CounterEnded payloads.
+            if (metric is CounterEndedPayload counterEnded)
+            {
+                if (_observedEndedCounters.Add((counterEnded.Provider, counterEnded.Name)))
+                {
+                    _logger.CounterEndedPayload(counterEnded.Name);
+                }
+                return;
+            }
+            if (metric is ErrorPayload errorPayload)
+            {
+                if (_observedErrorMessages.Add(errorPayload.ErrorMessage))
```

Comment:
Double check there is no provider/name here

Reply:
Also add comment regarding the above behavior.

Reply:
Verified that errors only have sessionId and errorMessage - Wiktor is going to add a comment.